Objects/unicodeobject.c

```diff
@@ -13885,6 +13886,153 @@ unicode_sizeof_impl(PyObject *self)
     return PyLong_FromSsize_t(size);
 }
 
+/*[clinic input]
+str.dedent as unicode_dedent
```

Comment:
I think it would be easier to maintain if we kept the implementation of this method in Python.  I suggest just keeping it in the textwrap module.  Just rename `textwrap.dedent` to `_dedent` and turn `dedent` into the public wrapper that adds the PendingDeprecationWarning.  That benefits other Python implementations (PyPy) who can continue using it and is nicer than this literal translation of the Python code into CPython API calls.

This C API version becomes a simple import of textwrap, lookup of dedent, and returning the result of the call.